### ISSUE TITLE
[explicit-module-boundary-types] allowArgumentsExplicitlyTypedAsAny doesn't works

### ISSUE THREAD

**Issue by llyp618** [typescript-eslint/typescript-eslint#2341]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/explicit-module-boundary-types": ["warn", {
      allowArgumentsExplicitlyTypedAsAny: false,
    }]
  }
}
```

```TS
 validate(payload: any): number {
    return 1;
  }
```

**Expected Result**
no warning
**Actual Result**
```
Argument 'payload' should be typed with a non-any type.eslint(@typescript-eslint/explicit-module-boundary-types)
```
**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `3.7.1` |
| `@typescript-eslint/parser`        | `3.7.1` |
| `TypeScript`                       | `3.9.3` |
| `ESLint`                           | `7.1.0` |
| `node`                             | `12.16.1` |
| `npm`                              | `6.13.4` |

**Comment by bradzacher** [comment]
have you got a more complete code sample?

We have a set of passing tests in master which suggest that this option is working
https://github.com/typescript-eslint/typescript-eslint/blob/817067b72b53c8f417f3c1c85e970ea82f0f6953/packages/eslint-plugin/tests/rules/explicit-module-boundary-types.test.ts#L586-L616

**Comment by llyp618** [comment]
Sorry, I made a mistake，This configuration takes effect
